### PR TITLE
Comply with go regexp & user defined regular expression prefix

### PR DESCRIPTION
**Pull request by Loyalsoldier**
To avoid further confusion like this: https://github.com/v2fly/domain-list-community/issues/163

### CODE REVIEW

**Comment by Loyalsoldier** [comment]
cc @kslr @iochen @EpLiar @Robot-DaneelOlivaw @IceCodeNew @MuenYu @studentmain

**Review by EpLiar — APPROVED** [review]

**Comment by studentmain** [comment]
I think we needn't touch it: domain rule is already not compatible with v2ray-core

**Comment by Robot-DaneelOlivaw** [comment]
If core can recognize the new expression, then it's fine to me. Ease confusion between config file and domain list at least.

File: README.md [hunk 1]
```diff
@@ -106,7 +106,7 @@ To generate a section:
 3. Omit all empty lines.
 4. Generate each `domain:` line into a [sub-domain routing rule](https://github.com/v2fly/v2ray-core/blob/master/app/router/config.proto#L21).
 5. Generate each `keyword:` line into a [plain domain routing rule](https://github.com/v2fly/v2ray-core/blob/master/app/router/config.proto#L17).
-6. Generate each `regex:` line into a [regex domain routing rule](https://github.com/v2fly/v2ray-core/blob/master/app/router/config.proto#L19).
+6. Generate each `regexp:` line into a [regex domain routing rule](https://github.com/v2fly/v2ray-core/blob/master/app/router/config.proto#L19).
```
[the remark this answers is not in the document]
Comment (Loyalsoldier) [reply README.md:111]: `regex domain routing rule` 里的 `regex` 是跟后面代码链接里的类型保持一致。

**Review by MuenYu — APPROVED** [review]

**Comment by Loyalsoldier** [comment]
@studentmain 

The notice:
> The following types of rules are **NOT** fully compatible with the ones that user defined in V2Ray config file. Do **Not** copy and paste directly.

**Comment by Loyalsoldier** [comment]
Merge due to majority approved. Please use `regexp:` as regular expression prefix in this repo in the future.

**Pull request merged**